### PR TITLE
jax.scipy.special: manually document functions to avoid scipy import

### PR DESCRIPTION
**Pull request by jakevdp**
This lets us give more implementation-specific information, and lets us avoid a dependency on scipy. Related to #20799.

Preview rendered docstrings here: https://jax--20822.org.readthedocs.build/en/20822/jax.scipy.html#module-jax.scipy.special

Part of #21461.

### CODE REVIEW

**Review by hawkinsp — COMMENTED** [review]
Same general comments on these:

* use type links where possible for `Array` and `ArrayLike`
* at least link to definition of the math in each case (Wikipedia being good), although having formulae is better where we can.
* an example would be peachy.

Doable?

**Comment by jakevdp** [comment]
It's all doable, it's just about how much time we want to invest

**Comment by jakevdp** [comment]
Mathematical definitions added

File: jax/_src/scipy/special.py [hunk 1]
```diff
@@ -32,131 +31,437 @@
...
+
+  .. math::
+
+     \mathrm{logit}(p) = \log\frac{p}{1 - p}
+
+  Args:
+    x: arraylike, real-valued.
+
+  Returns:
+    array contining values of the logit function.
```
Comment (hawkinsp) [inline jax/_src/scipy/special.py:417]: sp. containing

File: jax/_src/scipy/special.py [hunk 2]
```diff
@@ -32,131 +31,437 @@
...
 
 
-logsumexp = implements(osp_special.logsumexp, module='scipy.special')(ops_special.logsumexp)
+logsumexp = ops_special.logsumexp
 
 
 @custom_derivatives.custom_jvp
-@implements(osp_special.xlogy, module='scipy.special')
 def xlogy(x: ArrayLike, y: ArrayLike) -> Array:
+  """Compute x*log(y) with well-defined derivatives.
```
Comment (hawkinsp) [inline jax/_src/scipy/special.py:449]: What makes the derivatives well-defined?

You say below the primal value is defined at 0, 0, but what implications does that have for the derivative?

File: jax/_src/scipy/special.py [hunk 3]
```diff
@@ -171,8 +476,23 @@ def _xlogy_jvp(primals, tangents):
 
 
 @custom_derivatives.custom_jvp
-@implements(osp_special.xlog1py, module='scipy.special', update_doc=False)
 def xlog1py(x: ArrayLike, y: ArrayLike) -> Array:
+  """Compute x*log(1 + y) with well-defined derivatives.
```
Comment (hawkinsp) [inline jax/_src/scipy/special.py:480]: Same.

**Review by hawkinsp — APPROVED** [review]

File: jax/_src/scipy/special.py (hunk 2 above)
Comment (jakevdp) [reply jax/_src/scipy/special.py:449]: They're well-defined becuase the gradients are implemented in terms of `xlogy`

File: jax/_src/scipy/special.py (hunk 1 above)
Comment (jakevdp) [reply jax/_src/scipy/special.py]: Done

File: jax/_src/scipy/special.py (hunk 3 above)
Comment (jakevdp) [reply jax/_src/scipy/special.py]: Done

File: jax/_src/scipy/special.py (hunk 2 above)
Comment (jakevdp) [reply jax/_src/scipy/special.py]: Done
